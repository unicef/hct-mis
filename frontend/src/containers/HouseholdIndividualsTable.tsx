import React, { ReactElement, useState } from 'react';
import TableCell from '@material-ui/core/TableCell';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
import { HouseholdNode, IndividualNode } from '../__generated__/graphql';
import { Order, TableComponent } from '../components/table/TableComponent';
import { HeadCell } from '../components/table/EnhancedTableHead';
import { ClickableTableRow } from '../components/table/ClickableTableRow';
import { useBusinessArea } from '../hooks/useBusinessArea';
import { StatusBox } from '../components/StatusBox';
import { paymentRecordStatusToColor } from '../utils/utils';

const headCells: HeadCell<IndividualNode>[] = [
  {
    disablePadding: false,
    label: 'Individual ID',
    id: 'individualCaId',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Individual',
    id: 'fullName',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Status',
    id: 'status',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Role',
    id: 'role',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Sex',
    id: 'sex',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Date of Birth',
    id: 'dob',
    numeric: true,
  },
  {
    disablePadding: false,
    label: 'Employment / Education',
    id: 'employment',
    numeric: true,
  },
];
const StatusContainer = styled.div`
  width: 120px;
`;

interface HouseholdIndividualsTableProps {
  household: HouseholdNode;
}
export function HouseholdIndividualsTable({
  household,
}: HouseholdIndividualsTableProps): ReactElement {
  const history = useHistory();
  const businessArea = useBusinessArea();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const [orderBy, setOrderBy] = useState(null);
  const [orderDirection, setOrderDirection] = useState('asc');
  const handleClick = (row): void => {
    history.push(`/${businessArea}/population/household/individual/${row.id}`);
  };

  const allIndividuals = household.individuals.edges.map((edge) => edge.node);
  const totalCount = allIndividuals.length;
  return (
    <TableComponent<IndividualNode>
      title='Individuals in Household'
      data={allIndividuals.slice(
        page * rowsPerPage,
        page * rowsPerPage + rowsPerPage,
      )}
      renderRow={(row) => {
        return (
          <ClickableTableRow
            hover
            onClick={() => handleClick(row)}
            role='checkbox'
            key={row.id}
          >
            <TableCell align='left'>{row.individualCaId}</TableCell>
            <TableCell align='left'>{row.fullName}</TableCell>
            <TableCell align='left'>
              <StatusContainer>
                <StatusBox
                  status='no data'
                  statusToColor={paymentRecordStatusToColor}
                />
              </StatusContainer>
            </TableCell>
            <TableCell align='left'>empty</TableCell>
            <TableCell align='left'>{row.sex}</TableCell>
            <TableCell align='left'>
              {row.dob ? row.estimatedDob : row.dob}
            </TableCell>
            <TableCell align='left'>empty</TableCell>
          </ClickableTableRow>
        );
      }}
      headCells={headCells}
      rowsPerPageOptions={totalCount < 5 ? [totalCount] : [5, 10, 15]}
      rowsPerPage={totalCount > 5 ? rowsPerPage : totalCount}
      page={page}
      itemsCount={totalCount}
      handleChangePage={(event, newPage) => {
        setPage(newPage);
      }}
      handleChangeRowsPerPage={(event) => {
        setRowsPerPage(Number(event.target.value));
        setPage(0);
      }}
      handleRequestSort={(event, property) => {
        let direction = 'asc';
        if (property === orderBy) {
          direction = orderDirection === 'asc' ? 'desc' : 'asc';
        }
        setOrderBy(property);
        setOrderDirection(direction);
      }}
      orderBy={orderBy}
      order={orderDirection as Order}
    />
  );
}
