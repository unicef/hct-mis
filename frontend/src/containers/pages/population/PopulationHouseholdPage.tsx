import { Box } from '@material-ui/core';
import get from 'lodash/get';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import {
  ProgramNode,
  useAllProgramsForChoicesQuery,
  useHouseholdChoiceDataQuery,
} from '../../../__generated__/graphql';
import { LoadingComponent } from '../../../components/core/LoadingComponent';
import { PageHeader } from '../../../components/core/PageHeader';
import { PermissionDenied } from '../../../components/core/PermissionDenied';
import { HouseholdFilters } from '../../../components/population/HouseholdFilter';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBusinessArea } from '../../../hooks/useBusinessArea';
import { usePermissions } from '../../../hooks/usePermissions';
import { getFilterFromQueryParams } from '../../../utils/utils';
import { HouseholdTable } from '../../tables/population/HouseholdTable';

const initialFilter = {
  search: '',
  program: '',
  residenceStatus: '',
  admin2: '',
  householdSizeMin: '',
  householdSizeMax: '',
  orderBy: 'unicef_id',
  withdrawn: null,
};

export const PopulationHouseholdPage = (): React.ReactElement => {
  const { t } = useTranslation();
  const location = useLocation();

  const [filter, setFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const [appliedFilter, setAppliedFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );

  const businessArea = useBusinessArea();
  const permissions = usePermissions();

  const { data, loading } = useAllProgramsForChoicesQuery({
    variables: { businessArea },
    fetchPolicy: 'cache-and-network',
  });
  const {
    data: choicesData,
    loading: choicesLoading,
  } = useHouseholdChoiceDataQuery({
    variables: { businessArea },
    fetchPolicy: 'cache-first',
  });

  if (loading || choicesLoading) return <LoadingComponent />;
  if (permissions === null) return null;

  if (!hasPermissions(PERMISSIONS.POPULATION_VIEW_HOUSEHOLDS_LIST, permissions))
    return <PermissionDenied />;

  if (!choicesData) return null;

  const allPrograms = get(data, 'allPrograms.edges', []);
  const programs = allPrograms.map((edge) => edge.node);

  return (
    <>
      <PageHeader title={t('Households')} />
      <HouseholdFilters
        programs={programs as ProgramNode[]}
        filter={filter}
        choicesData={choicesData}
        setFilter={setFilter}
        initialFilter={initialFilter}
        appliedFilter={appliedFilter}
        setAppliedFilter={setAppliedFilter}
      />
      <Box
        display='flex'
        flexDirection='column'
        data-cy='page-details-container'
      >
        <HouseholdTable
          filter={appliedFilter}
          businessArea={businessArea}
          choicesData={choicesData}
          canViewDetails={hasPermissions(
            PERMISSIONS.POPULATION_VIEW_HOUSEHOLDS_DETAILS,
            permissions,
          )}
        />
      </Box>
    </>
  );
};
