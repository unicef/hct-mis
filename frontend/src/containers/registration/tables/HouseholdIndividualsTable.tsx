import React, {ReactElement, useState} from 'react';
import styled from 'styled-components';
import {useHistory} from 'react-router-dom';
import {ImportedIndividualMinimalFragment, useHouseholdChoiceDataQuery} from '../../../__generated__/graphql';
import {LoadingComponent} from '../../../components/LoadingComponent';
import {HeadCell} from "../../../components/table/EnhancedTableHead";
import {useBusinessArea} from "../../../hooks/useBusinessArea";
import {Order, TableComponent} from "../../../components/table/TableComponent";
import {ImportedIndividualsTableRow} from './ImportedIndividualsTable/ImportedIndividualsTableRow';

const headCells: HeadCell<ImportedIndividualMinimalFragment>[] = [
  {
    disablePadding: false,
    label: 'Individual ID',
    id: 'unicefId',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Individual',
    id: 'fullName',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Role',
    id: 'role',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Relationship to HoH',
    id: 'relationship',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Date of Birth',
    id: 'birthDate',
    numeric: true,
  },
  {
    disablePadding: false,
    label: 'Gender',
    id: 'sex',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Dedupe within Batch',
    id: 'deduplicationBatchStatus',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Dedupe against Population',
    id: 'deduplicationGoldenRecordStatus',
    numeric: false,
  },
];
const StatusContainer = styled.div`
  min-width: 120px;
  max-width: 200px;
`;

export function HouseholdImportedIndividualsTable({
  household,
}): ReactElement {
  const history = useHistory();
  const businessArea = useBusinessArea();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const [orderBy, setOrderBy] = useState(null);
  const [orderDirection, setOrderDirection] = useState('asc');

  const { data: choicesData, loading } = useHouseholdChoiceDataQuery();

  if (loading) return <LoadingComponent />;

  if (!choicesData) {
    return null;
  }

  const allIndividuals = household.individuals.edges.map((edge) => edge.node);
  if (orderBy) {
    if (orderDirection === 'asc') {
      allIndividuals.sort((a, b) => {
        return a[orderBy] < b[orderBy] ? 1 : -1;
      });
    } else {
      allIndividuals.sort((a, b) => {
        return a[orderBy] > b[orderBy] ? 1 : -1;
      });
    }
  }

  const totalCount = allIndividuals.length;
  return (
    <TableComponent<ImportedIndividualMinimalFragment>
      title='Individuals in Household'
      data={allIndividuals.slice(
        page * rowsPerPage,
        page * rowsPerPage + rowsPerPage,
      )}
      allowSort={false}
      renderRow={(row) => {
        return (
          <ImportedIndividualsTableRow choices={choicesData} individual={row} />
        );
      }}
      headCells={headCells}
      rowsPerPageOptions={totalCount < 5 ? [totalCount] : [5, 10, 15]}
      rowsPerPage={totalCount > 5 ? rowsPerPage : totalCount}
      page={page}
      itemsCount={totalCount}
      handleChangePage={(event, newPage) => {
        setPage(newPage);
      }}
      handleChangeRowsPerPage={(event) => {
        setRowsPerPage(Number(event.target.value));
        setPage(0);
      }}
      handleRequestSort={(event, property) => {
        let direction = 'asc';
        if (property === orderBy) {
          direction = orderDirection === 'asc' ? 'desc' : 'asc';
        }
        setOrderBy(property);
        setOrderDirection(direction);
      }}
      orderBy={orderBy}
      order={orderDirection as Order}
    />
  );
}
