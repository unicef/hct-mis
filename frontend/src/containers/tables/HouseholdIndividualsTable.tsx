import React, {ReactElement, useState} from 'react';
import TableCell from '@material-ui/core/TableCell';
import styled from 'styled-components';
import {useHistory} from 'react-router-dom';
import {HouseholdChoiceDataQuery, HouseholdNode, IndividualNode,} from '../../__generated__/graphql';
import {Order, TableComponent} from '../../components/table/TableComponent';
import {HeadCell} from '../../components/table/EnhancedTableHead';
import {ClickableTableRow} from '../../components/table/ClickableTableRow';
import {useBusinessArea} from '../../hooks/useBusinessArea';
import {StatusBox} from '../../components/StatusBox';
import {choicesToDict, populationStatusToColor, sexToCapitalize,} from '../../utils/utils';
import {UniversalMoment} from '../../components/UniversalMoment';

const headCells: HeadCell<IndividualNode>[] = [
  {
    disablePadding: false,
    label: 'Individual ID',
    id: 'unicefId',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Individual',
    id: 'fullName',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Status',
    id: 'status',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Role',
    id: 'role',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Relationship to HoH',
    id: 'relationship',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Date of Birth',
    id: 'birthDate',
    numeric: true,
  },
  {
    disablePadding: false,
    label: 'Gender',
    id: 'sex',
    numeric: false,
  },
];
const StatusContainer = styled.div`
  min-width: 120px;
  max-width: 200px;
`;

interface HouseholdIndividualsTableProps {
  household: HouseholdNode;
  choicesData: HouseholdChoiceDataQuery;
}
export function HouseholdIndividualsTable({
  household,
  choicesData,
}: HouseholdIndividualsTableProps): ReactElement {
  const history = useHistory();
  const businessArea = useBusinessArea();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const [orderBy, setOrderBy] = useState(null);
  const [orderDirection, setOrderDirection] = useState('asc');
  const handleClick = (row): void => {
    history.push(`/${businessArea}/population/individuals/${row.id}`);
  };

  const relationshipChoicesDict = choicesToDict(
    choicesData?.relationshipChoices,
  );
  const roleChoicesDict = choicesToDict(choicesData?.roleChoices);
  const allIndividuals = household.individuals.edges.map((edge) => edge.node);
  if (orderBy) {
    if (orderDirection === 'asc') {
      allIndividuals.sort((a, b) => {
        return a[orderBy] < b[orderBy] ? 1 : -1;
      });
    } else {
      allIndividuals.sort((a, b) => {
        return a[orderBy] > b[orderBy] ? 1 : -1;
      });
    }
  }

  const totalCount = allIndividuals.length;
  return (
    <TableComponent<IndividualNode>
      title='Individuals in Household'
      data={allIndividuals.slice(
        page * rowsPerPage,
        page * rowsPerPage + rowsPerPage,
      )}
      allowSort={false}
      renderRow={(row) => {
        return (
          <ClickableTableRow
            hover
            onClick={() => handleClick(row)}
            role='checkbox'
            key={row.id}
          >
            <TableCell align='left'>{row.unicefId}</TableCell>
            <TableCell align='left'>{row.fullName}</TableCell>
            <TableCell align='left'>
              <StatusContainer>
                <StatusBox
                  status={row.status}
                  statusToColor={populationStatusToColor}
                />
              </StatusContainer>
            </TableCell>
            <TableCell align='left'>{roleChoicesDict[row.role]}</TableCell>
            <TableCell align='left'>
              {household?.id === row?.household?.id
                ? relationshipChoicesDict[row.relationship]
                : relationshipChoicesDict.NON_BENEFICIARY}
            </TableCell>
            <TableCell align='left'>
              <UniversalMoment>{row.birthDate}</UniversalMoment>
            </TableCell>
            <TableCell align='left'>{sexToCapitalize(row.sex)}</TableCell>
          </ClickableTableRow>
        );
      }}
      headCells={headCells}
      rowsPerPageOptions={totalCount < 5 ? [totalCount] : [5, 10, 15]}
      rowsPerPage={totalCount > 5 ? rowsPerPage : totalCount}
      page={page}
      itemsCount={totalCount}
      handleChangePage={(event, newPage) => {
        setPage(newPage);
      }}
      handleChangeRowsPerPage={(event) => {
        setRowsPerPage(Number(event.target.value));
        setPage(0);
      }}
      handleRequestSort={(event, property) => {
        let direction = 'asc';
        if (property === orderBy) {
          direction = orderDirection === 'asc' ? 'desc' : 'asc';
        }
        setOrderBy(property);
        setOrderDirection(direction);
      }}
      orderBy={orderBy}
      order={orderDirection as Order}
    />
  );
}
