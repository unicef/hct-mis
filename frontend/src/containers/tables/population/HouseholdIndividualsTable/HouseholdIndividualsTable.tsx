import TableCell from '@mui/material/TableCell';
import { ReactElement, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { BlackLink } from '@components/core/BlackLink';
import { StatusBox } from '@components/core/StatusBox';
import { ClickableTableRow } from '@components/core/Table/ClickableTableRow';
import { HeadCell } from '@components/core/Table/EnhancedTableHead';
import { Order, TableComponent } from '@components/core/Table/TableComponent';
import { UniversalMoment } from '@components/core/UniversalMoment';
import {
  choicesToDict,
  populationStatusToColor,
  sexToCapitalize,
} from '@utils/utils';
import {
  HouseholdChoiceDataQuery,
  HouseholdNode,
  IndividualNode,
} from '@generated/graphql';
import { useBaseUrl } from '@hooks/useBaseUrl';

const headCells: HeadCell<IndividualNode>[] = [
  {
    disablePadding: false,
    label: 'Individual ID',
    id: 'unicefId',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Individual',
    id: 'fullName',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Status',
    id: 'status',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Role',
    id: 'role',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Relationship to HoH',
    id: 'relationship',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Date of Birth',
    id: 'birthDate',
    numeric: true,
  },
  {
    disablePadding: false,
    label: 'Gender',
    id: 'sex',
    numeric: false,
  },
];

interface HouseholdIndividualsTableProps {
  household: HouseholdNode;
  choicesData: HouseholdChoiceDataQuery;
}
export function HouseholdIndividualsTable({
  household,
  choicesData,
}: HouseholdIndividualsTableProps): ReactElement {
  const navigate = useNavigate();
  const { baseUrl } = useBaseUrl();
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const [orderBy, setOrderBy] = useState(null);
  const [orderDirection, setOrderDirection] = useState('asc');
  const handleClick = (row): void => {
    navigate(`/${baseUrl}/population/individuals/${row.id}`);
  };

  const relationshipChoicesDict = choicesToDict(
    choicesData?.relationshipChoices,
  );
  const roleChoicesDict = choicesToDict(choicesData?.roleChoices);
  const allIndividuals = household?.individuals?.edges?.map(
    (edge) => edge.node,
  );
  if (orderBy) {
    if (orderDirection === 'asc') {
      allIndividuals.sort((a, b) => (a[orderBy] < b[orderBy] ? 1 : -1));
    } else {
      allIndividuals.sort((a, b) => (a[orderBy] > b[orderBy] ? 1 : -1));
    }
  }

  const totalCount = allIndividuals.length;
  return (
    <TableComponent<IndividualNode>
      title="Individuals in Household"
      data={allIndividuals.slice(
        page * rowsPerPage,
        page * rowsPerPage + rowsPerPage,
      )}
      allowSort={false}
      renderRow={(row) => (
        <ClickableTableRow
          hover
          onClick={() => handleClick(row)}
          role="checkbox"
          key={row.id}
        >
          <TableCell align="left">
            <BlackLink to={`/${baseUrl}/population/individuals/${row.id}`}>
              {row.unicefId}
            </BlackLink>
          </TableCell>
          <TableCell align="left">{row.fullName}</TableCell>
          <TableCell align="left">
            <StatusBox
              status={row.status}
              statusToColor={populationStatusToColor}
            />
          </TableCell>
          <TableCell align="left">{roleChoicesDict[row.role]}</TableCell>
          <TableCell align="left">
            {household?.id === row?.household?.id
              ? relationshipChoicesDict[row.relationship]
              : relationshipChoicesDict.NON_BENEFICIARY}
          </TableCell>
          <TableCell align="left">
            <UniversalMoment>{row.birthDate}</UniversalMoment>
          </TableCell>
          <TableCell align="left">{sexToCapitalize(row.sex)}</TableCell>
        </ClickableTableRow>
      )}
      headCells={headCells}
      rowsPerPageOptions={totalCount < 5 ? [totalCount] : [5, 10, 15]}
      rowsPerPage={totalCount > 5 ? rowsPerPage : totalCount}
      page={page}
      itemsCount={totalCount}
      handleChangePage={(event, newPage) => {
        setPage(newPage);
      }}
      handleChangeRowsPerPage={(event) => {
        setRowsPerPage(Number(event.target.value));
        setPage(0);
      }}
      handleRequestSort={(event, property) => {
        let direction = 'asc';
        if (property === orderBy) {
          direction = orderDirection === 'asc' ? 'desc' : 'asc';
        }
        setOrderBy(property);
        setOrderDirection(direction);
      }}
      orderBy={orderBy}
      order={orderDirection as Order}
    />
  );
}
