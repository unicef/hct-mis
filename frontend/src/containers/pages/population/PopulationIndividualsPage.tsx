import { Box } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import {
  useHouseholdChoiceDataQuery,
  useIndividualChoiceDataQuery,
} from '@generated/graphql';
import { LoadingComponent } from '@components/core/LoadingComponent';
import { PageHeader } from '@components/core/PageHeader';
import { PermissionDenied } from '@components/core/PermissionDenied';
import { IndividualsFilter } from '@components/population/IndividualsFilter';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { usePermissions } from '@hooks/usePermissions';
import { getFilterFromQueryParams } from '@utils/utils';
import { IndividualsListTable } from '../../tables/population/IndividualsListTable';

export function PopulationIndividualsPage(): React.ReactElement {
  const { t } = useTranslation();
  const location = useLocation();
  const { businessArea } = useBaseUrl();
  const permissions = usePermissions();
  const { data: householdChoicesData, loading: householdChoicesLoading } =
    useHouseholdChoiceDataQuery();

  const { data: individualChoicesData, loading: individualChoicesLoading } =
    useIndividualChoiceDataQuery();

  const initialFilter = {
    search: '',
    documentType: individualChoicesData?.documentTypeChoices?.[0]?.value,
    documentNumber: '',
    admin2: '',
    sex: '',
    ageMin: '',
    ageMax: '',
    flags: [],
    orderBy: 'unicef_id',
    status: '',
    lastRegistrationDateMin: '',
    lastRegistrationDateMax: '',
  };


  const [filter, setFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const [appliedFilter, setAppliedFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );

  if (householdChoicesLoading || individualChoicesLoading)
    return <LoadingComponent />;

  if (!individualChoicesData || !householdChoicesData || permissions === null)
    return null;

  if (
    !hasPermissions(PERMISSIONS.POPULATION_VIEW_INDIVIDUALS_LIST, permissions)
  )
    return <PermissionDenied />;

  return (
    <>
      <PageHeader title={t('Individuals')} />
      <IndividualsFilter
        filter={filter}
        choicesData={individualChoicesData}
        setFilter={setFilter}
        initialFilter={initialFilter}
        appliedFilter={appliedFilter}
        setAppliedFilter={setAppliedFilter}
      />
      <Box
        display="flex"
        flexDirection="column"
        data-cy="page-details-container"
      >
        <IndividualsListTable
          filter={appliedFilter}
          businessArea={businessArea}
          choicesData={householdChoicesData}
          canViewDetails={hasPermissions(
            PERMISSIONS.POPULATION_VIEW_INDIVIDUALS_DETAILS,
            permissions,
          )}
        />
      </Box>
    </>
  );
}
