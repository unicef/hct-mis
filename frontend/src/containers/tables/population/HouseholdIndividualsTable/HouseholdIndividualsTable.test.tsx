import React from 'react';
import { fakeHousehold } from '../../../../../fixtures/population/fakeHousehold';
import { fakeHouseholdChoices } from '../../../../../fixtures/population/fakeHouseholdChoices';
import { render } from '../../../../testUtils/testUtils';
import { HouseholdIndividualsTable } from './HouseholdIndividualsTable';

describe('components/tables/population/HouseholdIndividualsTable', () => {
  it('should render', () => {
    const { container } = render(
      <HouseholdIndividualsTable
        household={fakeHousehold}
        choicesData={fakeHouseholdChoices}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
