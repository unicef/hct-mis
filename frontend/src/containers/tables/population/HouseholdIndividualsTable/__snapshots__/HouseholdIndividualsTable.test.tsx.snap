// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/tables/population/HouseholdIndividualsTable should render 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root"
      >
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-root-8 MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
              data-cy="table-title"
            >
              Individuals in Household
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-11"
          />
        </div>
        <table
          aria-label="enhanced table"
          aria-labelledby="tableTitle"
          class="MuiTable-root makeStyles-table-3"
          data-cy="table-title"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Individual ID
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Individual
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Status
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Role
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Relationship to HoH
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Date of Birth
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  class="sc-bZQynM fgJQBP"
                  data-cy="table-label"
                >
                  Gender
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root sc-bxivhb ksDfcC MuiTableRow-hover"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-bdVaJa frsrLc"
                  href="/afghanistan/population/individuals/SW5kaXZpZHVhbE5vZGU6YTI2ODYyMTAtYjBhNi00NTI1LWI4YTAtYTFmY2FlMWIwMjE5"
                >
                  IND-42-0000.0001
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Alicja Kowalska
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-bwzfXH jDYBWd"
                >
                  <div
                    class="sc-htpNat dNIAUP"
                    data-cy="status-container"
                  >
                    ACTIVE
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Alternate collector
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Not a Family Member. Can only act as a recipient.
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <time
                  datetime="-894672000000"
                >
                  26 Aug 1941
                </time>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Female
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-bxivhb ksDfcC MuiTableRow-hover"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-bdVaJa frsrLc"
                  href="/afghanistan/population/individuals/SW5kaXZpZHVhbE5vZGU6ZjVmMzgwN2ItYTBjOS00Mjk0LTg0Y2QtNDhkMjllMjZhODU3"
                >
                  IND-88-0000.0005
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Agata Kowalska
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-bwzfXH jDYBWd"
                >
                  <div
                    class="sc-htpNat dNIAUP"
                    data-cy="status-container"
                  >
                    ACTIVE
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Primary collector
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Head of household (self)
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <time
                  datetime="-188611200000"
                >
                  10 Jan 1964
                </time>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Female
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-bxivhb ksDfcC MuiTableRow-hover"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-bdVaJa frsrLc"
                  href="/afghanistan/population/individuals/SW5kaXZpZHVhbE5vZGU6YWI5YWZhNzUtMTg0Yy00NDIxLWJmMGUtNTIzZGUxNWMyOGFm"
                >
                  IND-88-0000.0006
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Angela Kowalska
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-bwzfXH jDYBWd"
                >
                  <div
                    class="sc-htpNat dNIAUP"
                    data-cy="status-container"
                  >
                    ACTIVE
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                None
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Brother / Sister
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <time
                  datetime="947462400000"
                >
                  10 Jan 2000
                </time>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                Female
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
        data-cy="table-pagination"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-3 of 3
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
